test/Microsoft.DotNet.ProjectJsonMigration.Tests/MigrationBackupPlanTests.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
```

Comment:
I'd like to see a test which uses a global.json file to define the solution root. I believe these tests assume the root is the parent directory of the starting project.

It would also be useful to have a test which does a restore of some root directory. E.g.

```
/core
  /github
    /piotrp
      /cli
        global.json
```

In the above, I should be able to run migrate from any directory above `global.json` and have migration pick up the root directory as the directory that contains global.json

Finally, I'm not sure if global.json's `src` property supported relative paths to a parent directory. If it did, it would be good to understand how we handle those since the `backup` directory's layout cannot handle `..`'s afaik.